contracts/Aavegotchi/facets/ItemsFacet.sol

```diff
@@ -182,38 +190,76 @@ contract ItemsFacet is Modifiers {
     ///@dev A wearable cannot be equipped in the wrong slot
     ///@param _tokenId The identifier of the aavegotchi to make changes to
     ///@param _wearablesToEquip An array containing the identifiers of the wearables to equip
-    function equipWearables(uint256 _tokenId, uint16[EQUIPPED_WEARABLE_SLOTS] calldata _wearablesToEquip)
+    function equipWearables(
+        uint256 _tokenId,
+        uint16[EQUIPPED_WEARABLE_SLOTS] calldata _wearablesToEquip
+    )
         external
+    {
+        uint256[EQUIPPED_WEARABLE_SLOTS] memory _commitmentIds;
+        _equipWearables(_tokenId, _wearablesToEquip, _commitmentIds);
+    }
+
+    ///@notice Allow the owner of a claimed aavegotchi to equip/unequip wearables to his aavegotchi
+    ///@dev Only valid for claimed aavegotchis
+    ///@dev A zero value will unequip that slot and a non-zero value will equip that slot with the wearable whose identifier is provided
+    ///@dev A wearable cannot be equipped in the wrong slot
+    ///@param _tokenId The identifier of the aavegotchi to make changes to
+    ///@param _wearablesToEquip An array containing the identifiers of the wearables to equip
+    ///@param _commitmentIds An array containing the identifiers of the deposited wearables to equip
+    function equipDelegatedWearables(
+        uint256 _tokenId,
+        uint16[EQUIPPED_WEARABLE_SLOTS] calldata _wearablesToEquip,
+        uint256[EQUIPPED_WEARABLE_SLOTS] calldata _commitmentIds
+    )
+        external
+    {
+        
+        _equipWearables(_tokenId, _wearablesToEquip, _commitmentIds);
+    }
+
+
+    function _equipWearables(
+        uint256 _tokenId,
+        uint16[EQUIPPED_WEARABLE_SLOTS] calldata _wearablesToEquip,
+        uint256[EQUIPPED_WEARABLE_SLOTS] memory _commitmentIdsToEquip
+    )
+        internal
         onlyAavegotchiOwner(_tokenId)
         onlyUnlocked(_tokenId)
     {
         Aavegotchi storage aavegotchi = s.aavegotchis[_tokenId];
         require(aavegotchi.status == LibAavegotchi.STATUS_AAVEGOTCHI, "LibAavegotchi: Only valid for AG");
-        emit EquipWearables(_tokenId, aavegotchi.equippedWearables, _wearablesToEquip);
-
-        address sender = LibMeta.msgSender();
+        emit EquipWearables(_tokenId, s.aavegotchis[_tokenId].equippedWearables, _wearablesToEquip);
+        emit EquipDelegatedWearables(_tokenId, s.gotchiEquippedItemsInfo[_tokenId].equippedCommitmentIds, _commitmentIdsToEquip);
+        GotchiEquippedCommitmentsInfo storage _gotchiInfo = s.gotchiEquippedItemsInfo[_tokenId];
 
         for (uint256 slot; slot < EQUIPPED_WEARABLE_SLOTS; slot++) {
+            
             uint256 toEquipId = _wearablesToEquip[slot];
             uint256 existingEquippedWearableId = aavegotchi.equippedWearables[slot];
+            bool _sameWearablesIds = toEquipId == existingEquippedWearableId;
```

Comment:
Can you explain the checks that are happening in this section? @ernanirst

Reply:
In the old `equipWearables` function, all we needed was to check if the tokenIds were the same in order to decide if we can skip it (if the tokenIds are the same, there is no actual change on that wearable slot).

However, in the new `equipDelegatedWearables` function, the user might replace a delegated wearable with a wearable they own (or vice-versa). So we need to check both the tokenIds and the commitmentIds to know if we can skip it (line 248). We also assign this comparison to a variable because it's also used on line 258, to treat a different edge case